src/dnx.coreclr.unix/dnx.coreclr.cpp

```diff
@@ -233,6 +237,113 @@ int InvokeDelegate(host_main_fn host_main, int argc, const char** argv)
     return exit_code;
 }
 
+#if defined(PLATFORM_LINUX)
+
+std::string get_os_version()
+{
+    std::vector<std::string> qualifiers { "DISTRIB_ID=", "DISTRIB_RELEASE=" };
+
+    std::ifstream lsb_release;
+    lsb_release.open("/etc/lsb-release", std::ifstream::in);
+    if (lsb_release.is_open())
+    {
+        std::string os_version;
+        for (std::string line; std::getline(lsb_release, line); )
+        {
+            for (auto& qualifier : qualifiers)
+            {
+                if (line.compare(0, qualifier.length(), qualifier) == 0)
+                {
+                    if (os_version.length() > 0)
+                    {
+                        os_version.append(" ");
+                    }
+                    os_version.append(line.substr(qualifier.length()));
+                }
+            }
+        }
+
+        if (os_version.length() == 0)
+        {
+            fprintf(stderr, "Could not find version information. OS version will default to the empty string.\n");
+        }
+
+        return os_version;
+    }
+
+    fprintf(stderr, "Could not open /etc/lsb_release. OS version will default to the empty string.\n");
+    return "";
+}
+#else
+std::string translate_darwin_version(const std::string release)
+{
+    auto dot_position = release.find(".");
+    if (dot_position == std::string::npos)
+    {
+        fprintf(stderr, "Could not determine os version.\n");
+        return "10.1";
+    }
+
+    auto version = stoi(release.substr(0, dot_position));
+    return std::string("10.").append(std::to_string(version - 4));
```

Comment:
Yeah, I found a similar table on Wikipedia. In theory, we should be using the table but since it's always `10.(Darwin Version - 4)`, this approach seems fine in practice.